p256/src/arithmetic/projective.rs

```diff
@@ -194,16 +193,37 @@ impl ProjectivePoint {
 
     /// Returns `[k] self`.
     fn mul(&self, k: &Scalar) -> ProjectivePoint {
-        let mut ret = ProjectivePoint::IDENTITY;
-
-        for limb in k.limbs().iter().rev() {
-            for i in (0..Limb::BIT_SIZE).rev() {
-                ret = ret.double();
-                ret.conditional_assign(&(ret + self), Choice::from(((limb.0 >> i) & 1) as u8));
+        let mut pc = [ProjectivePoint::default(); 16];
+        pc[0] = ProjectivePoint::IDENTITY;
+        pc[1] = *self;
+        for i in 2..16 {
+            pc[i] = if i % 2 == 0 {
+                pc[i / 2].double()
+            } else {
+                pc[i - 1].add(self)
+            };
+        }
+        let mut q = ProjectivePoint::IDENTITY;
+        let k_ = k.to_bytes();
+        let k = k_.as_slice();
+        let mut pos = 256 - 4;
+        loop {
+            let slot = (k[31 - (pos >> 3) as usize] >> (pos & 7)) & 0xf;
```

Comment:
Was there actually an issue here using the `Index` impl on `GenericArray`? i.e. why doesn't this work:
```suggestion
        let k = k.to_bytes();
        let mut pos = 256 - 4;
        loop {
            let slot = (k[31 - (pos >> 3) as usize] >> (pos & 7)) & 0xf;
```

Reply:
Sure!